bluepysnap/config.py

```diff
@@ -132,6 +131,40 @@ def resolve(self):
         return self._resolve(self.content)
 
     @staticmethod
-    def parse(filepath):
+    def parse(config, configdir):
         """Parse SONATA network config."""
-        return Config(filepath).resolve()
+        return Parser(config, configdir).resolve()
+
+
+class Config:
+    """Common config class."""
+
+    def __init__(self, config, config_class):
+        """Initializes the Config class.
+
+        Args:
+            config (str): Path to the configuration file
+            config_class (class): libsonata class corresponding to the configuration file, either
+                libsonata.CircuitConfig or libsonata.SimulationConfig
+        """
+        self._configdir = str(Path(config).parent.absolute())
+        self._libsonata = config_class.from_file(config)
+
+    @classmethod
+    def from_circuit_config(cls, config_path):
+        """Instantiate the config class from circuit configuration."""
+        return cls(config_path, libsonata.CircuitConfig)
+
+    @classmethod
+    def from_simulation_config(cls, config_path):
+        """Instantiate the config class from simulation configuration."""
+        return cls(config_path, libsonata.SimulationConfig)
+
+    def to_dict(self):
+        """Return the configuration as a dict with absolute paths."""
+        if hasattr(self._libsonata, "expanded_json"):
+            config = json.loads(self._libsonata.expanded_json)
+        else:
+            config = json.loads(self._libsonata.json)
```

Comment:
Out of curiosity, what is the diff between the two?

Reply:
`expanded_json` is currently only available in `libsonata.CircuitConfig` and it expands the manifest values in the config, whereas `json` is just the raw json data as it is in the (in this case simulation) configuration file.

Reply:
I have fixed this; both `CircuitConfig` and `SimulationConfig` should be the same now:
https://github.com/BlueBrain/libsonata/pull/214

Reply:
I'll add this `SimulationConfig.expanded_json`, too.

Due to my short-sightedness on the matter, though, I just realized I can not remove the manifest resolution in the `Parse` class as it is needed by the validation code. :man_facepalming: 

However, when we go forward with the schema based validation for the config file in #155 , I'd _assume_ this would then no longer be needed.